tests/test_cli_compile.py

```diff
@@ -534,3 +534,23 @@ def test_cert_option(MockPyPIRepository, runner, option, attr, expected):
     for call in MockPyPIRepository.call_args_list:
         pip_options = call[0][0]
         assert getattr(pip_options, attr) == expected
+
+
+@pytest.mark.parametrize('option, expected', [
+    ('--build-isolation', True),
+    ('--no-build-isolation', False),
+])
+@mock.patch('piptools.scripts.compile.PyPIRepository')
+def test_build_isolation_option(MockPyPIRepository, runner, option, expected):
+    """
+    A value of the --build-isolation/--no-build-isolation flag must be passed to the PyPIRepository.
+    """
+    with open('requirements.in', 'w'):
+        pass
+
+    runner.invoke(cli, [option])
+
+    # Ensure the build_isolation in PyPIRepository has have the expected option
+    for call in MockPyPIRepository.call_args_list:
+        build_isolation = call[0][2]
+        assert build_isolation is expected
```

Comment:
I'd prefer to assert/compare the whole list, but maybe only still the single option.  This would include the number of calls then also - currently it might just assert nothing silently.

Reply:
Sorry didn't get your point. Could you shoot some code-snippet for the instance?

Reply:
1. `MockPyPIRepository.call_args_list` might be empty

```python
assert [call[0][2] for call in MockPyPIRepository.call_args_list] == [expected, …
```]

Reply:
btw: you could use `[ci skip]` for the commit with the suggested changes - since it should be squashed later anyway, and would save some hours of CI.

Reply:
Do you mean something like `MockPyPIRepository.assert_called_once_with(..., ..., expected)`?

Reply:
Yes, or the above, where you expect it to be a non-empty list.

Reply:
> btw: you could use [ci skip] for the commit with the suggested changes - since it should be squashed later anyway, and would save some hours of CI.

Right, okay.

Reply:
Or we just could assert it called once (before iteration over call_args_list) to ensure the call_args_list is not empty:
```python
MockPyPIRepository.assert_called_once()
```

Reply:
In that case `test_cert_option` also should be fixed.

Reply:
> Or we just could assert it called once (before iteration over call_args_list) to ensure the call_args_list is not empty:
> 
> ```python
> MockPyPIRepository.assert_called_once()
> ```

Yes.  Although I would prefer a single assertion using a list comprehension instead.

Reply:
I was thinking, you know, one assertion looks better. Fixed the test.

Reply:
Yes, one assertion is better in my opinion, too.

`assert [call[0][2] for call in MockPyPIRepository.call_args_list] == [expected]` is just that, no? :)

Thanks for adjusting!